### DISCUSSION TITLE
runPrimal for FSI simulation

### DISCUSSION THREAD

**Discussion by erkhans** [mdolab/dafoam#693] [General]
Hello. 
When I use runPrimal in the transient simulation, it creates only one .f5 file. How can I make it to create more .f5 files to see the deformation? I have attached the runScript.txt file.
![image](https://github.com/user-attachments/assets/ee3ccdac-40c4-4a28-930b-78a685a22b4b)

[runScript.txt](https://github.com/user-attachments/files/17324262/runScript.txt)

**Comment by friedenhe** [comment]
You can use the DAFoam's built-in `f5tovtk` util to convert f5 to vtk.

**Reply by erkhans** [reply]
Thank you, Professor. The problem is that when I use "runPrimal" for 2-way FSI, it does not deform the structure. How can I enable it?

**Comment by friedenhe** [comment]
OK, the current version does not support transient FSI...

**Reply by erkhans** [reply]
Thank you, professor. How do I enable it, if I do static FSI simulation?

**Comment by friedenhe** [comment]
Did you mean to "enable structure deformation in paraview post-processing"? We covered this in the 2022 workshop. Check the slides and video.

**Reply by erkhans** [reply]
No, I mean when we do aerostructural optimization, the structure deforms. But when we do FSI simulation with runPrimal, the structure does not deform. It just shows the contours. I have attached the screenshot. How I can make the structure deform during FSI simulation?
![opt](https://github.com/user-attachments/assets/a9e217ff-b18c-4187-9d88-1fbafbe48655)
![fsi](https://github.com/user-attachments/assets/5e0d57c2-fc3d-4686-b847-8e911a663049)

**Comment by friedenhe** [comment]
Can you elaborate on how do you check if the structure deforms? Can you reproduce this issue with the MACH-tutorial wing case?

**Reply by erkhans** [reply]
When I visualize the results from optimization (using opt) in paraview, I can see the deformations of structure as in the first picture above. However, when I visualize the results from FSI (using runPrimal) in paraview, it just shows the loadings on the structure, and there is no deformation of the structure. It is shown in the second figure above.  But, when running the FSI simulation, there should be deformation. What needs to be done to make the structure deform in FSI simulation just like in an optimization simulation? Could you, please, help me with that? I have also attached the runScript.
[runScript.txt](https://github.com/user-attachments/files/17829724/runScript.txt)

**Comment by friedenhe** [comment]
OK, what I meant was: which file did you load for the 1st and 2nd pictures? I guess you used cruise_00x for the 1st and cruise_000 for the 2nd. The reason you saw deformation from the 1st pic was because it was changed by the FFD point displacement during optimization, instead of the structural deformation. Anyway, you need to carefully read the DAFoam workshop 2022 about how to visualize structural deformation in paraview (you need to use the warp by scalar feature).
